lib/service_pub_rabbit.js

```diff
@@ -21,50 +22,32 @@ class ServicePubRabbit extends Component.mixin(AsyncEmitter) {
         this._initializing = false;
         this._connection = null;
         this._namespace = config.default("service.rabbit.namespace", "default");
+        this._connectionConfig = config.default("service.rabbit", { host: "localhost", port: 5672 });
+
+        this._eventList = [];
+        const publishedEvent = this.config.default("service.publish", {});
+        for (const eventName in publishedEvent) {
+            this._eventList.push(eventName);
+        }
+        this._publishedEvent = publishedEvent;
 
         this.servicePub.on("trigger", this.publishEvent.bind(this));
-        this.amqp.on("connected", () => {
-            this.init();
-        });
     }
 
     *initialize() {
-        if (this.amqp.getConnection())
-            this.init();
     }
 
-    *init() {
-        if (this._initializing) return;
-        this._initializing = true;
-
-        this._connection = this.amqp.getConnection();
-        this._eventList = [];
-        this._channels = {};
-
-        let desc = this.config.default("service.publish", {});
-        for (let i in desc) {
-            this._eventList.push(i);
-            yield this.createPublisher(i, desc[i]);
-        }
-
-        this._initializing = false;
+    *destroy() {
     }
 
     publishEvent(event, payload) {
-        let channel = this._channels[event];
-        if (channel && this._eventList.includes(event)) {
-            let exchangeName = this._namespace + "." + this.SERVICE_NAME + "." + event;
-            let content = JSON.stringify(payload);
-            channel.publish(exchangeName, "", Buffer.from(content), { persistent: true });
-        }
-    }
-
-    *createPublisher(event) {
-        let channel = yield this._connection.createChannel();
-        let exchangeName = this._namespace + "." + this.SERVICE_NAME + "." + event;
-        this._channels[event] = channel;
-
-        yield channel.assertExchange(exchangeName, "fanout", { durable: true });
+        const rabbit = new Rabbit({
```

Comment:
with this approach, we will instantiate `Rabbit` every time we publish message?

Reply:
should it better to instantiate `Rabbit` class first on constructor before we use it on this method?

Reply:
or maybe make it singleton?